.github/actions/clp-core-build-containers/action.yaml

```diff
@@ -0,0 +1,91 @@
+name: "clp-core-build-containers"
+description: "Builds and publishes container images for CLP-core's dependencies and binaries"
+
+inputs:
+  os_name:
+    description: "Name of container OS"
+    required: true
+  clp_core_dir:
+    description: "CLP-core's component directory"
+    required: true
+  push_deps_image:
+    description: "Whether to publish a container image containing CLP's dependencies"
+    required: true
+  push_binaries_image:
+    description: "Whether to publish a container image containing CLP's binaries"
+    required: true
+  local_registry_port:
+    description: "Port number for the local registry"
+    required: true
+  token:
+    description: "Registry token"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - uses: "docker/setup-buildx-action@v3"
+      with:
+        driver-opts: "network=host"
+
+    - if: "inputs.push_deps_image == 'true'"
+      uses: "docker/login-action@v3"
+      with:
+        registry: "ghcr.io"
+        username: "${{github.actor}}"
+        password: "${{inputs.token}}"
+
+    - id: "get_deps_image_props"
+      shell: "bash"
+      run: |
+        if [[ "${{inputs.push_deps_image}}" == "true" ]] ; then
+          registry=ghcr.io
+          branch=${{github.ref_name}}
+        else
+          registry=localhost:${{inputs.local_registry_port}}
+          branch=latest
+        fi
```

Comment:
Is there no way to do this using github action's if? I know there is no else condition so maybe it's not as clean? 
Also, is there a benefit to assigning registry and branch in the if/else, as opposed to just echoing inside the if/else?

Reply:
I can't think of any way to do this directly with GitHub actions atm:
* If we introduced two steps which were keyed with the `if` and `else` conditions, they would need different `id`s, so that wouldn't work for the next few steps.
* We could maybe simplify the bash statement using GH actions' [ternary-like behaviour](https://docs.github.com/en/actions/learn-github-actions/expressions#example), but to set the vars for all the steps to use it, we'd still have to use `$GITHUB_OUPUT`. Between GH actions' syntax and bash's syntax, I think bash's syntax is probably more consistent / less confusing.

Reply:
Good point about the simplifying the variable setting. Fixed.